src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -695,8 +699,9 @@ bool IsComInterop (IMarshalInfoProvider marshalInfoProvider, TypeReference param
 			return false;
 		}
 
-		void ProcessCreateInstanceByName (in MessageOrigin origin, bool diagnosticsEnabled, MethodDefinition calledMethod, ValueNodeList methodParams)
+		void ProcessCreateInstanceByName (in MessageOrigin origin, in DiagnosticContext diagnosticContext, MethodDefinition calledMethod, ValueNodeList methodParams)
```

Comment:
Why this case is special and has both `origin` and `diagnosticContext` and not just `diagnosticContext`?

Reply:
No good reason, I meant to remove this assert but forgot about it. Good catch!